terraform/modules/apps/publishing-api/main.tf

```diff
@@ -7,82 +7,161 @@ terraform {
-  name        = "fargate_${var.service_name}_ingress"
-  vpc_id      = var.vpc_id
-  description = "Permit internal services to access the ${var.service_name} ECS service"
+data "aws_secretsmanager_secret" "router_api_bearer_token" {
+  name = "publishing_api_app-ROUTER_API_BEARER_TOKEN"
 }
 
-resource "aws_security_group" "dependencies" {
-  name        = "fargate_${var.service_name}_app"
-  vpc_id      = var.vpc_id
-  description = "Allows ingress from ${var.service_name} to its dependencies"
+data "aws_secretsmanager_secret" "secret_key_base" {
+  name = "publishing_api_app-SECRET_KEY_BASE"
+}
 
-  egress {
-    from_port   = 0
-    to_port     = 0
-    protocol    = "-1"
-    cidr_blocks = ["0.0.0.0/0"]
-  }
+data "aws_secretsmanager_secret" "sentry_dsn" {
+  name = "publishing_api_app-SENTRY_DSN"
 }
 
-#
-# Dependencies
-#
 
-data "aws_security_group" "service_dependencies" {
-  id = "sg-05ad7398fc0d7c5b4" # legacy govuk-aws group: govuk_publishing-api_access
+module "app" {
+  source                           = "../../app"
+  cpu                              = "512"
+  memory                           = "1024"
+  vpc_id                           = var.vpc_id
+  cluster_id                       = var.cluster_id
+  service_name                     = var.service_name
+  subnets                          = var.private_subnets
+  mesh_name                        = var.mesh_name
+  service_discovery_namespace_id   = var.service_discovery_namespace_id
+  service_discovery_namespace_name = var.service_discovery_namespace_name
+  task_role_arn                    = var.task_role_arn
+  execution_role_arn               = var.execution_role_arn
+  extra_security_groups            = [var.govuk_management_access_security_group]
+  container_definitions = [
+    {
+      "name" : "publishing-api",
+      "image" : "govuk/publishing-api:with-content-schemas",
+      "essential" : true,
+      "environment" : [
+        # TODO: factor our hardcoded stuff
+        { "name" : "APPMESH_VIRTUAL_NODE_NAME", "value" : "mesh/${var.mesh_name}/virtualNode/${var.service_name}" },
+        { "name" : "CONTENT_API_PROTOTYPE", "value" : "yes" },
+        { "name" : "CONTENT_STORE", "value" : "http://content-store.${var.service_discovery_namespace_name}" },
+        { "name" : "DRAFT_CONTENT_STORE", "value" : "https://draft-content-store.${var.service_discovery_namespace_name}" },
+        { "name" : "EVENT_LOG_AWS_ACCESS_ID", "value" : "AKIAJE6VSW25CYBUMQJA" },
+        { "name" : "EVENT_LOG_AWS_BUCKETNAME", "value" : "govuk-publishing-api-event-log-test" },
+        { "name" : "EVENT_LOG_AWS_USERNAME", "value" : "govuk-publishing-api-event-log_user" },
+        { "name" : "DEFAULT_TTL", "value" : "1800" },
+        { "name" : "GOVUK_APP_DOMAIN", "value" : var.service_discovery_namespace_name },
+        { "name" : "GOVUK_APP_DOMAIN_EXTERNAL", "value" : var.govuk_app_domain_external },
+        { "name" : "GOVUK_APP_NAME", "value" : var.service_name },
```

Comment:
Is this intentionally set to var.service_name? We haven't done this for content-store or publisher so if it makes sense to set GOVUK_APP_NAME based on service_name then we should update the others to match and make sure the description service_name mentions that it affects app config as well as AWS resources.

I haven't extensively looked into all the ways that GOVUK_APP_NAME is used but it looks like it's used for things like redis key prefixes.

Reply:
I think it is a good idea to not repeat the service name over and over. Makes it easier to change later and/orprevent typos.

Reply:
It's not really a question of avoiding repetition in this case, it's a question of semantics. In other words, if we were to have more than one instance of this module in the same deployment, should GOVUK_APP_NAME be the same for both or different? If it should be different, then setting it to var.service_name seems reasonable. In that case, we should do the same for the other ones.

Reply:
yeah, I agree with that. I think we have not crossed that road yet. I can change it back.